### PR TITLE
[nrf noup] include: net: add socket option level IPPROTO_ALL

### PR DESCRIPTION
**Pull request by MarkusLassila**
Add socket option level IPPROTO_ALL to socket_ncs.h.

PR that depends on this: https://github.com/nrfconnect/sdk-nrf/pull/11715

### CODE REVIEW

File: include/zephyr/net/socket_ncs.h [hunk 1]
```diff
@@ -121,6 +112,26 @@ extern "C" {
...
+
+/** IPv4 and IPv6 protocol level (pseudo-val) for nRF sockets. */
+#define IPPROTO_ALL 512
+/** sockopt: disable all replies to unexpected traffics */
+#define SO_SILENCE_ALL 30
+
+/* NCS specific IPPROTO_IP level socket options */
+
+/** sockopt: disable IPv4 ICMP replies */
+#define SO_IP_ECHO_REPLY 31
```
Comment (lemrey) [inline include/zephyr/net/socket_ncs.h:126]: Non-zero enables ICMP echo replies on IPv4. Perhaps this should be "enable IPv4 ICMP replies"? (same goes for the IPv6 socket option)

Comment (MarkusLassila) [reply include/zephyr/net/socket_ncs.h]: Changed: disable -> enable

**Comment by MarkusLassila** [comment]
> 

Thank you, I tried to adapt to the comment style, but missed this. Is there somewhere where the comment guideline would be documented?

**Review by rlubos — APPROVED** [review]

**Review by lemrey — APPROVED** [review]
Looks good. It's documented [here ](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/releases_and_maturity/developing/code_base.html#oss-repositories-downstream-project-history)

**Pull request merged**